grpc/src/main/java/io/stargate/grpc/service/Service.java

```diff
@@ -742,6 +764,9 @@ private Connection newConnection(AuthenticatedUser user) {
       connection = persistence.newConnection();
     }
     connection.login(user);
+    if (user.token() != null) {
```

Comment:
Why do we check for token to be present here? Would we not want process users that were authenticated by other means?

Reply:
The persistence API can only check authorization if a token is present: https://github.com/stargate/stargate/blob/master/persistence-cassandra-3.11/src/main/java/io/stargate/db/cassandra/impl/StargateQueryHandler.java#L212

Reply:
True, but would it be possible to have unconditional logic for injecting user data into `ClientState` and let `Persistence` deal with specific user details?